### DISCUSSION TITLE
[Idea/Improvement] Update script

### DISCUSSION THREAD

**Discussion by GilOliveira** [joaomlourenco/novathesis#373] [Ideas]
It may be an interesting idea to create a script, like in Python for instance, that automatically keeps novathesis up-to-date.
It could detect the version the user is running (from line 11 of `template.tex`), fetch the latest repo, create a backup folder with the legacy files, and merge the new template code with the thesis text, while maintaining the `.git` folder of the old version (in case the user is using it for their own version control).

An alternative would be something using git submodules but that would probably require more changes to the way the template files are organized.

I could probably try to implement something like this myself but my thesis is not letting me have much more free time and I recon some people here should be more proficient in scripting to implement this in the fraction of the time, so, if something is looking for improvement ideas to implement, there can be something interesting here.

**Comment by joaomlourenco** [comment]
@GilOliveira, that's a great idea!  I have done it (partially)!  The plan was to help people upgrading from v4 to v5 (there were many deep changes in the major upgrade).  But I never really concluded ti to the point of being bullet proof… and never shared it!

But if someone wants to take the lead, I will gladly share the few scripts I have… maybe some of them are useful as a starting point!
